plug-ins/sslstrip/sslstrip.c

```diff
@@ -667,537 +765,553 @@ static int http_get_peer(struct http_connection *connection)
 #ifndef OS_LINUX
 static size_t http_create_ident(void **i, struct packet_object *po)
 {
-	struct http_ident *ident;
+   struct http_ident *ident;
 
-	SAFE_CALLOC(ident, 1, sizeof(struct http_ident));
+   SAFE_CALLOC(ident, 1, sizeof(struct http_ident));
 
-	ident->magic = HTTP_MAGIC;
+   ident->magic = HTTP_MAGIC;
 
-	memcpy(&ident->L3_src, &po->L3.src, sizeof(struct ip_addr));
-	ident->L4_src = po->L4.src;
-	ident->L4_dst = po->L4.dst;
+   memcpy(&ident->L3_src, &po->L3.src, sizeof(struct ip_addr));
+   ident->L4_src = po->L4.src;
+   ident->L4_dst = po->L4.dst;
 
-	/* return the ident */
-	*i = ident;
-	return sizeof(struct http_ident);
+   /* return the ident */
+   *i = ident;
+   return sizeof(struct http_ident);
 }
 #endif
 
 static int http_sync_conn(struct http_connection *connection) 
 {
-	if (http_get_peer(connection) != E_SUCCESS)
-		return -E_INVALID;
+   if (http_get_peer(connection) != E_SUCCESS)
+      return -E_INVALID;
 
 
-	set_blocking(connection->fd, 0);
-	return E_SUCCESS;
+   set_blocking(connection->fd, 0);
+   return E_SUCCESS;
 }
 
 static int http_read(struct http_connection *connection, struct packet_object *po)
 {
-	int len = 0;	
+   int len = 0;   
 
-	len = read(connection->fd, po->DATA.data, HTTP_MAX);
+   len = read(connection->fd, po->DATA.data, HTTP_MAX);
 
-	po->DATA.len = len;
+   po->DATA.len = len;
 
-	if(len <= 0)
-		return -E_INVALID;
+   if(len <= 0)
+      return -E_INVALID;
 
-	return len;	
+   return len;   
 }
 
 static void http_handle_request(struct http_connection *connection, struct packet_object *po)
 {
-	struct https_link *link;
+   struct https_link *link;
 
-	SAFE_CALLOC(connection->request->url, 1, 512);
+   SAFE_CALLOC(connection->request->url, 1, 512);
 
-	if (connection->request->url==NULL)
-		return;
+   if (connection->request->url==NULL)
+      return;
 
-	Find_Url(po->DATA.data, &connection->request->url);
+   Find_Url(po->DATA.data, &connection->request->url);
 
-	if (connection->request->url == NULL) {
-		return;
-	}
+   if (connection->request->url == NULL) {
+      return;
+   }
 
 
-	//parse HTTP request
-	if (!memcmp(po->DATA.data, "GET", 3)) {
-		connection->request->method = HTTP_GET;
-	} else if (!memcmp(po->DATA.data, "POST", 4)) {
-		connection->request->method = HTTP_POST;
-	}
+   //parse HTTP request
+   if (!memcmp(po->DATA.data, "GET", 3)) {
+      connection->request->method = HTTP_GET;
+   } else if (!memcmp(po->DATA.data, "POST", 4)) {
+      connection->request->method = HTTP_POST;
+   }
 
-	char *r = (char*)po->DATA.data;
+   char *r = (char*)po->DATA.data;
 
-	//Skip the first line of request
-	r = strstr((const char*)po->DATA.data, "\r\n");
-	r += 2; //Skip \r\n
+   //Skip the first line of request
+   r = strstr((const char*)po->DATA.data, "\r\n");
+   r += 2; //Skip \r\n
 
-	char *h = strdup(r);
-	char *body = strdup(r);
-	BUG_IF(h==NULL);
-	BUG_IF(body==NULL);
+   char *h = strdup(r);
```

Comment:
to reproduce:
build ettercap debug mode, uncomment redir commands on iptables.

gw target 1
virtualbox windows 8,1 machine target2 (bridge mode, promiscuous mode allowed)
sniff remote connections, arp spoofing, enable sslstrip.

on windows open IE (also firefox, but navigating more than one ssl website in the same session gives the ssl error, because one ssl fingerprint can't be used for two different websites)

I went to 
https://alioth.debian.org/
http://ssl.rustedmusic.com/
https://ssl.rustedmusic.com/
and after some f5 ettercap crashed.

I admit, the websites didn't open (there was the connecting progress bar, but nothing showed, at least until ettercap crashed)
https://dl.dropboxusercontent.com/u/35375391/build.tar.xz
the full build directory, this way you can (I hope) gdb it easily, since it contains also the libettercap.so file

BTW it is almost similar to the previous one...